### PR TITLE
TST: skips in extension.test_categorical

### LINKED ISSUE #38904: TST: skips in extension.test_categorical
ref: https://github.com/pandas-dev/pandas/pull/38901#discussion_r550899659

At least some of the tests that are skipped here should instead be xfailed.

### PR DESCRIPTION
**Pull request by rhshadrach**
- [x] closes #38904  
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

cc @simonjayhawkins 

For the searchsorted test, the categorical data starts off ordered as "C" < "A" < "B" but then gets reordered in the test by these lines:

> b, c, a = data_for_sorting
        arr = type(data_for_sorting)._from_sequence([a, b, c])

so the ordering becomes "A" < "B" < "C". This causes failure when "C" is said to be inserted at index 3 (new order) to preserve order rather than the expected 0 (original order).

### CODE REVIEW

File: pandas/tests/extension/test_categorical.py [hunk 1]
```diff
@@ -175,9 +175,8 @@ def test_combine_add(self, data_repeated):
     def test_fillna_length_mismatch(self, data_missing):
         super().test_fillna_length_mismatch(data_missing)
 
-    def test_searchsorted(self, data_for_sorting):
-        if not data_for_sorting.ordered:
-            raise pytest.skip(reason="searchsorted requires ordered data.")
+    def test_searchsorted(self, data_for_sorting, as_series):
+        super().test_searchsorted(data_for_sorting, as_series)
```
Comment (simonjayhawkins) [inline pandas/tests/extension/test_categorical.py:179]: could remove entirely if no need to override? but maybe just update to xfail depending on base test change discussion.

File: pandas/tests/extension/base/methods.py [hunk 2]
```diff
@@ -394,7 +394,10 @@ def test_hash_pandas_object_works(self, data, as_frame):
 
     def test_searchsorted(self, data_for_sorting, as_series):
         b, c, a = data_for_sorting
-        arr = type(data_for_sorting)._from_sequence([a, b, c])
+        # Pass dtype so sorting order is not lost for Categorical
+        arr = type(data_for_sorting)._from_sequence(
+            [a, b, c], dtype=data_for_sorting.dtype
+        )
```
Comment (simonjayhawkins) [inline pandas/tests/extension/base/methods.py:400]: not sure if we should be changing base extensions tests. @jorisvandenbossche?

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @rhshadrach for following up on this.

File: pandas/tests/extension/base/methods.py (hunk 2 above)
Comment (jorisvandenbossche) [reply pandas/tests/extension/base/methods.py:400]: I think that's fine to change it here (all EAs should be able to handle a passed dtype to `_from_sequence`, so shouldn't influence the test for others)

Now, it can probably also be done in a shorter way, something like `arr = data_for_sorting.take([2, 0, 1])  # to get [a, b, c]`

File: pandas/tests/extension/test_categorical.py (hunk 1 above)
Comment (jorisvandenbossche) [reply pandas/tests/extension/test_categorical.py:179]: Yes, can be removed entirely

File: pandas/tests/extension/base/methods.py (hunk 2 above)
Comment (rhshadrach) [reply pandas/tests/extension/base/methods.py:400]: Thanks @jorisvandenbossche, will update with the shortened form.

File: pandas/tests/extension/test_categorical.py [hunk 3]
```diff
@@ -229,21 +228,26 @@ def test_consistent_casting(self, dtype, expected):
...
-        if op_name != "__rmod__":
-            super().test_arith_frame_with_scalar(data, all_arithmetic_operators)
-        else:
-            pytest.skip("rmod never called when string is first argument")
-
-    def test_arith_series_with_scalar(self, data, all_arithmetic_operators):
-
+        if op_name == "__rmod__":
+            request.node.add_marker(
+                pytest.mark.xfail(
```
Comment (jorisvandenbossche) [inline pandas/tests/extension/test_categorical.py:236]: Is this change related? (I wouldn't know if an xfail is more appropriate here than a skip)

Comment (rhshadrach) [reply pandas/tests/extension/test_categorical.py:236]: Related to #38904 that this PR is trying to close, but not to the other skip/xfail in this PR. Will split off into a separate PR if desired.

This test is doing op(Series, const) expecting it to raise, where Series consists of and const is a string.  There is no rmod in the python library `operator`, and so the test is instead doing `lambda x, y: mod(const, Series)` . This doesn't raise.

This is always expected to "fail" (meaning not raise), and I think we'd want to know if it didn't. This is why I went with xfail over skip here.

File: pandas/tests/extension/base/methods.py [hunk 4]
```diff
@@ -394,7 +394,8 @@ def test_hash_pandas_object_works(self, data, as_frame):
 
     def test_searchsorted(self, data_for_sorting, as_series):
         b, c, a = data_for_sorting
-        arr = type(data_for_sorting)._from_sequence([a, b, c])
+        # Pass dtype so sorting order is not lost for Categorical
```
Comment (simonjayhawkins) [inline pandas/tests/extension/base/methods.py:397]: no need for this comment now.

Comment (rhshadrach) [reply pandas/tests/extension/base/methods.py]: Doh - thanks for catching this.

**Comment by jreback** [comment]
thanks @rhshadrach
